tests/destinations/test_path_utils.py

```diff
@@ -51,19 +127,168 @@ def test_get_table_prefix_layout() -> None:
 
     # disallow missing table_name
     with pytest.raises(CantExtractTablePrefix):
-        path_utils.get_table_prefix_layout("some_random{schema_name}/stuff_in_between/")
+        get_table_prefix_layout("some_random{schema_name}/stuff_in_between/")
 
     # disallow other params before table_name
     with pytest.raises(CantExtractTablePrefix):
-        path_utils.get_table_prefix_layout("{file_id}some_random{table_name}/stuff_in_between/")
+        get_table_prefix_layout("{file_id}some_random{table_name}/stuff_in_between/")
 
     # disallow any placeholders before table name (ie. Athena)
     with pytest.raises(CantExtractTablePrefix):
-        path_utils.get_table_prefix_layout(
+        get_table_prefix_layout(
             "{schema_name}some_random{table_name}/stuff_in_between/",
             supported_prefix_placeholders=[],
         )
 
     # disallow table_name without following separator
     with pytest.raises(CantExtractTablePrefix):
-        path_utils.get_table_prefix_layout("{schema_name}/{table_name}{load_id}.{file_id}.{ext}")
+        get_table_prefix_layout("{schema_name}/{table_name}{load_id}.{file_id}.{ext}")
+
+
+def test_create_path_uses_provided_load_package_timestamp(test_load: TestLoad) -> None:
+    load_id, job_info = test_load
+    now = pendulum.now()
+    timestamp = str(int(now.timestamp()))
+    path = create_path(
+        "{schema_name}/{table_name}/{load_id}.{file_id}.{timestamp}.{ext}",
+        schema_name="schema_name",
+        load_id=load_id,
+        load_package_timestamp=now.to_iso8601_string(),
+        file_name=job_info.file_name(),
+    )
+
+    assert timestamp in path
+    assert path.endswith(f"{timestamp}.{job_info.file_format}")
+
+
+def test_create_path_resolves_current_datetime(test_load: TestLoad) -> None:
+    """Check the flow when the current_datetime is passed
+
+    Happy path checks
+
+    1. Callback,
+    2. pendulum.DateTime
+
+    Failures when neither callback nor the value is not of pendulum.DateTime
+    """
+    load_id, job_info = test_load
+    now = pendulum.now()
+    now_timestamp = now.to_iso8601_string()
+    calls = 0
+
+    def current_datetime_callback():
+        nonlocal calls
+        calls += 1
+        return now
+
+    create_path(
```

Comment:
weird that you are not testing substrings with datetime in returned paths

Reply:
Done.